src/test/ui/build-hasher-object-safe.rs

```diff
@@ -0,0 +1,10 @@
+// run-pass
+
+use std::hash::BuildHasher;
+use std::collections::hash_map::{DefaultHasher, RandomState};
+
+fn ensure_object_safe(_: &dyn BuildHasher<Hasher = DefaultHasher>) {}
+
+fn main() {
+    ensure_object_safe(&RandomState::new());
+}
```

Comment:
No need to make this a ui test I think; a test in `library/core/tests` should suffice.